### PR TITLE
feat(ui-shell): add divider element and styles

### PR DESCRIPTION
**Pull request by aledavila**
Adds divider so users are able to use it where they want

#### Changelog

**Changed**

- switcher is an element now with its own styling so that users are able to place it where they want

#### Testing / Reviewing. 

Check Switcher

### CODE REVIEW

**Review by joshblack — COMMENTED** [review]
Should this be its own custom component instead of the `hr` tag? Also, would this impact anyone currently using side nav who are expecting the dividers to automatically render?

**Comment by aledavila** [comment]
@joshblack yea I agree I'll make it a component. It would affect it (if they are using it). The problem is that its set to be after every first then every 4th. and renders as an :after selector and isn't customizable at all.

**Review by asudoh — APPROVED** [review]
LGTM :+1: - Thanks @aledavila!

File: packages/react/src/components/UIShell/README.md [hunk 1]
```diff
@@ -22,3 +21,8 @@
     - `SideNavLink`: renders a link in the side nav
     - `SideNavMenu`: renders a collapsible menu in the side nav
       - `SideNavMenuItem`: renders a link in a side nav menu
+  - `RightPanel`: used for render the container for header actions on the right
+    - `Switcher`: used to render a list of product links inside the right panel
+      - `SwitcherItem`: used to render list item
+        - `SwitcherItemLink`: used to render a link inside the list item
+      - `Swithcer Divder`: used to render a divider for list items
```
Comment (emyarod) [inline packages/react/src/components/UIShell/README.md]:
```suggestion
      - `SwitcherDivider`: used to render a divider for list items
```

**Review by emyarod — APPROVED** [review]
looks good to me, pending design approval

**Comment by designertyler** [comment]
Switcher items should be 32px tall.
image

**Comment by designertyler** [comment]
The divider should be 224px wide. 
Screen Shot 2019-06-20 at 11 04 52 AM

**Review by designertyler — COMMENTED** [review]
Divider looks good. Thanks for merging 🎉

**Pull request merged**